docs/content/en/latest/architecture/supported_frameworks/nextjs.md

```diff
@@ -0,0 +1,144 @@
+---
+title: Render GoodData Visualizations in Next.js
+linkTitle: Next.js
+copyright: (C) 2007-2018 GoodData Corporation
+---
+
+Next.js is not supported out-of-the-box by GoodData.UI.
+
+However, rendering the GoodData visualizations in your Next.js project is not as daunting as it seems. This article serves as a quick demo of how to integrate GoodData visualizations into your project.
+### Step 1: Install all necessary dependencies
+
+The following packages deal with GoodData backend:
+
+```bash
+npm i @gooddata/api-client-tiger @gooddata/sdk-backend-tiger
+```
+
+The following package contains *all* components of GoodData UI SDK:
+
+```bash
+npm i @gooddata/sdk-ui-all
+```
+
+{{% alert title="Install only some components"%}}
+If you do not need to install all components, you can select only those you need. For more information refer to the [documentation](../../../learn/integrate_and_authenticate/cn_and_cloud_integration/#IntegrateCNandCloudintoanExistingApplication-Step1Installthenecessarydependencies).
+{{% /alert %}}
+
+The following package will help you generate a file with code representation of all available metrics and attributes in your GoodData Cloud:
+
+```bash
+npm i -D @gooddata/catalog-export
+```
+
+### Step 2: Extend package.json
+
+You need to modify `package.json`, and add the following properties:
+
+```bash
+"gooddata": {
+    "hostname": "<your-gooddata-host>",
+    "workspaceId": "<your-workspace-id>",
+    "catalogOutput": "app/gooddata-export.ts",
+    "backend": "tiger"
+}
+```
+
+This configuration is for the `catalog-export` tool, which exports all available metrics and attributes to a single file that you can reference in your code.
+
+### Step 3: Add a new script to package.json
+
+```bash
+"scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint",
+    "refresh-md": "gdc-catalog-export" <-- this is the new script you need to add!
+},
+```
+
+### Step 4: Add API token
+
+```bash
+export TIGER_API_TOKEN=<your-api-token>
+```
+
+*Note: more info on how to get API token is in the [documentation](https://www.gooddata.com/developers/cloud-native/doc/cloud/getting-started/create-api-token/).*
+
+### Step 5: Configure CORS in GoodData
+
+If your Next.js application runs on a different domain than GoodData (which is the most probable scenario), you need to configure CORS. Basically, you need to go to `<your-gooddata-host>/settings` and add Cross-origin resource sharing (CORS). If your Next.js application runs on domain *https://super-cool-app.com*, you need to add *https://super-cool-app.com* to CORS. You can find more information in the [documentation](https://www.gooddata.com/developers/cloud-native/doc/cloud/manage-organization/set-up-cors-for-organization/).
```

Comment:
If your Next.js application runs on a different domain than GoodData (which is the most likely scenario), you'll need to configure CORS (Cross-Origin Resource Sharing). To do this, navigate to `<your-gooddata-host>/settings` and add your Next.js domain to the CORS settings. For example, if your Next.js application is hosted at https://super-cool-app.com, you would need to add *https://super-cool-app.com* to the CORS settings in GoodData. For more details, you can consult the [GoodData documentation](https://www.gooddata.com/developers/cloud-native/doc/cloud/manage-organization/set-up-cors-for-organization/).